src/modules/imaging/camera.py

```diff
@@ -64,15 +66,24 @@ class WebcamCamera(CameraProvider):
     """
     Debug camera source which uses the computer's webcam as the image source.
     """
+    def __init__(self):
+        self.cap = cv2.VideoCapture(0)  # 0 is typically the default webcam
+        self.size = (640, 480)  # default size
+        self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, self.size[0])
+        self.cap.set(cv2.CAP_PROP_FRAME_HEIGHT, self.size[1])
 
     def set_size(self, size: Tuple[int, int]):
-        # TODO
-        raise NotImplementedError()
+        self.size = size
+        self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, size[0])
+        self.cap.set(cv2.CAP_PROP_FRAME_HEIGHT, size[1])
 
     def capture(self) -> Image.Image:
-        # TODO
-        raise NotImplementedError()
-
+        ret, frame = self.cap.read()
+        if ret:
+            frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
```

Comment:
We should resize the image here. Seems like opencv doesn't guarantee that the size is *strictly* the same as requested. Try changing the sample script as so:

```diff
--- a/samples/camera.py
+++ b/samples/camera.py
@@ -7,6 +7,8 @@ from src.modules.imaging.camera import WebcamCamera
 
 cam = WebcamCamera()
 
+cam.set_size((1000, 1000))
+
 print("Taking a picture in 3s.")
 
 for i in reversed(range(3)):
 ```